orthanc/orthanc-anon/plugin/pixl.py

```diff
@@ -248,10 +249,14 @@ def _process_dicom_instance(receivedDicom: bytes) -> tuple[orthanc.ReceivedInsta
 
     # Attempt to anonymise and drop the study if any exceptions occur
     try:
+        study_identifiers = get_study_info(dataset)
         anonymise_and_validate_dicom(dataset)
         return orthanc.ReceivedInstanceAction.MODIFY, write_dataset_to_bytes(dataset)
     except PixlDiscardError as error:
-        logger.debug("Skipping instance: {}", error)
+        logger.warning("Skipping {}: {}", study_identifiers, error)
+        return orthanc.ReceivedInstanceAction.DISCARD, None
+    except:  # noqa: E722 Allow bare except
```

Comment:
Yeah its a bit unorthdox but because this is being run in an orthanc callback, any uncaught exception won't take down the system anyway. Want to be explicit that we'll always discard any instance if there has been an exception

 There's the chance of different exceptions from the database query, hasher-api call, DICOM validation library and DICOM authentication library.

Reply:
ah I see, thanks. Yeah a comment might be good here